flow360/component/simulation/models/volume_models.py

```diff
@@ -0,0 +1,128 @@
+from typing import Optional
+
+import pydantic as pd
+
+from flow360.component.simulation.framework.base_model import Flow360BaseModel
+from flow360.component.simulation.models.material import Air, MaterialType
+from flow360.component.simulation.models.solver_numerics import (
+    NavierStokesSolver,
+    TransitionModelSolver,
+    TurbulenceModelSolverType,
+)
+
+
+class VolumeModelBase(Flow360BaseModel):
+    """
+    Base class for equation models
+
+    """
+
+    volumes: EntityList[Union[GenericVolume, str]] = pd.Field()
+    material: MaterialType = pd.Field()
+    initial_conditions: Optional[dict] = pd.Field(None)
+
+
+class FluidDynamicsAddOnBase(Flow360BaseModel):
+    volumes: EntityList[Union[GenericVolume, str]] = pd.Field()
+
+
+class FluidDynamics(VolumeModelBase):
+    """
+    General FluidDynamics volume model that contains all the common fields every fluid dynamics zone should have.
+    """
+
+    navier_stokes_solver: Optional[NavierStokesSolver] = pd.Field(None)
+    turbulence_model_solver: Optional[TurbulenceModelSolverType] = pd.Field(None)
+    transition_model_solver: Optional[TransitionModelSolver] = pd.Field(None)
+
+    material: MaterialTypes = pd.Field(Air())
+
+
+class HeatTransfer(VolumeModelBase):
+    """
+    General HeatTransfer volume model that contains all the common fields every heat transfer zone should have.
+    """
+
+    heat_equation_solver: Optional[components.HeatEquationSolver] = pd.Field(None)
+    volumetric_heat_source: Union[NonNegativeFloat, StrictStr] = pd.Field(0)
+
+
+class ActuatorDisk(FluidDynamicsAddOnBase):
+    """Same as Flow360Param ActuatorDisks.
+    Note that `center`, `axis_thrust`, `thickness` can be acquired from `entity` so they are not required anymore.
+    """
+
+    volumes: Optional[EntityList[Cylinder]] = pd.Field(None)
+
+    center: Optional[Coordinate] = pd.Field(None)
+    thickness: Optional[PositiveFloat] = pd.Field(None)
+    axis_thrust: Optional[Axis] = pd.Field(None)
+    force_per_area: ForcePerArea = pd.Field()
+
+
+class BETDisk(FluidDynamicsAddOnBase):
+    """Same as Flow360Param BETDisk.
+    Note that `center_of_rotation`, `axis_of_rotation`, `radius`, `thickness` can be acquired from `entity` so they are not required anymore.
+    """
+
+    volumes: Optional[EntityList[Cylinder]] = pd.Field(None)
+
+    rotation_direction_rule: Literal["leftHand", "rightHand"] = pd.Field("rightHand")
+    center_of_rotation: Optional[Coordinate] = pd.Field(None)
+    axis_of_rotation: Optional[Axis] = pd.Field(None)
+    thickness: Optional[LengthType.Positive] = pd.Field(None)
+    radius: Optional[LengthType.Positive] = pd.Field(None)
+    number_of_blades: pd.conint(strict=True, gt=0, le=10) = pd.Field()
+    omega: AngularVelocityType.NonNegative = pd.Field()
```

Comment:
why is this non-negative? Is it only controlled by `rotation_direction_rule` ?

Reply:
Yes, this is supposed to be speed